### DISCUSSION TITLE
Convert PNG to black/white byte array

### DISCUSSION THREAD

**Discussion by mu88** [dlemstra/Magick.NET#1404] [Help] [answered]
Hi 👋🏻

First of all, thank you for your awesome library!

I have a PNG image like this:
![image](https://github.com/dlemstra/Magick.NET/assets/4560672/7617c12b-67d4-461e-b760-2aa8580eac65)

To display it on an ePaper display, I need it as a black/white image. Furthermore, I only need a single `byte` per pixel (representing black or white as `0` resp. `255`), not a value per channel per pixel.

I did the b/w conversion like this:
```c#
using var image = new MagickImage();
await image.ReadAsync(screenshotFile);

if (blackAndWhite)
{
    image.Alpha(AlphaOption.Off);
    image.Threshold(new Percentage(95));
}

return image.ToByteArray();
```

That looks okay, but not as crisp like in image processing software like GIMP.

Now I have two questions:
1. [In the docs](https://www.imagemagick.org/Usage/quantize/#monochrome), I found the `monochrome` option. Where do I find this option in Magick.NET?
2. With my code, I still have an image with three channels, where every pixel of every channel is either `0` or `65536`. Of course I could manually process all pixels, discard two channels and replace `65536` by `255`.  
But is there a smarter way using one of Magick.NET's APIs for this task?

Thank you!

Versions:
- .NET 7, running on Linux (`amd64` and `arm64`)
- `<PackageReference Include="Magick.NET-Q16-AnyCPU" Version="13.1.3" />`

**Comment by dlemstra** [comment, marked as answer]
You are pretty close. Below is an example of what you could do to export the pixels instead:

```C#
using var image = new MagickImage();
image.Read(screenshotFile);

if (blackAndWhite)
{
    // Maybe gets you a better result ?
    image.Alpha(AlphaOption.Off);
    image.Threshold(new Percentage(95));

    // This will make your image monochrome.
    image.ColorType = ColorType.Bilevel;
}

// This will export the pixels of the image.
using var pixels = image.GetPixelsUnsafe();
pixels.ToByteArray("R");
```

**Reply by mu88** [reply]
thx a lot! So the `monochrome` option does not explicitly exist in Magick.NET's API, am I getting it correctly?

**Reply by dlemstra** [reply]
The code behind `-monochrome` does this: `image.ColorType = ColorType.Bilevel;`. But I did not want to add an extra method to the API for this.

**Reply by mu88** [reply]
Ah okay, got it 💪🏻
